nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -76,19 +148,129 @@ def __init__(self):
+            )
+            pos = joint_states.position
+            names = joint_states.name
+            names_pos_dict = {}
+            for i in range(len(names)):
+                names_pos_dict[names[i]] = pos[i]
+
+            self._right_joint_states = []
+            self._left_joint_states = []
+
+            # probably a redundant check but making sure the joint names exist from the /joint_states data
+            if not (
+                all(key in names_pos_dict.keys() for key in self.right_arm_joint_names)
+                and all(
+                    key in names_pos_dict.keys() for key in self.left_arm_joint_names
+                )
+            ):
+                rospy.logerr(
+                    "Invalid or missing joint name in data received from /joint_states"
+                )
+                return False
+
+            for i in range(7):
+                self._right_joint_states.append(
+                    names_pos_dict[self.right_arm_joint_names[i]]
+                )
+                self._left_joint_states.append(
+                    names_pos_dict[self.left_arm_joint_names[i]]
+                )
+
+            return True
+        except rospy.ROSException:
+            rospy.logwarn(
+                "Unable to get data from /joint_states, wait_for_message timeout"
+            )
+            return False
+
+    def get_joint_states(self):
+        if self.update_joint_states():
+            return {
+                "left_arm": self._left_joint_states,
+                "right_arm": self._right_joint_states,
+            }
+        return False
+
+    def verify_joint_target(self, joint_target: list, joint_states: list) -> bool:
+        """
+        Verifies that the joint target list is valid, and that it is not the same as the current joint state
+
+        @param joint_target:
+        @param joint_states:
+        @return: True if any targets are different from the current state,
+                 False if all targets are the same as current state
+        """
+        if not self.update_joint_states():
+            return False
+
+        if len(joint_target) != 7:
+            raise Exception("Invalid joint value length")
+        error_count = 7
+        for i in range(7):
+            if math.isclose(
+                abs(joint_target[i]),
+                abs(joint_states[i]),
+                rel_tol=self._joint_tolerance,
+            ):
+                error_count -= 1
+        return bool(error_count)
+
+    def cmd_right_arm(self, joint_target: list, blocking=True) -> bool:
+        """
+        Commands movement of the right arm through the right_arm action server
+
+        @param joint_target: list of target joint values
+        @param blocking: default True, if blocking, function will wait for action server response
+        @return: result of verify_joint_target and action server state if blocking
+        """
+        if self.verify_joint_target(joint_target, self._right_joint_states):
```

Comment:
Nit: Instead of nesting if statements, use early returns with inverted conditions
`if not {condition1} return\n
if not {condition2} return\n
goal  = ...\n
send_goal()`